interoperator/controllers/multiclusterdeploy/provisioner/provisioner_controller.go

```diff
@@ -93,6 +109,10 @@ func (r *ReconcileProvisioner) Reconcile(req ctrl.Request) (ctrl.Result, error)
 	clusterID := clusterInstance.GetName()
 	log.Info("reconciling cluster", "clusterID", clusterID)
 
+	// Setting the cluster state metric as Down.
+	// The cluster is ready when the reconcile completes.
+	clusterMetric.WithLabelValues(clusterID).Set(0)
```

Comment:
Hi @anoopjb , setting this value in the prometheus client itself would show the cluster to down, for a small amount of time(till reconcile sets it up again). 
Should we set the state in an inmemory variable and after reconcilePrimaryClusterIdConfig() method, we set the state of cluster to up or down in prometheus client based on error or success?